refactor(parser): Improve Transfer Family models with examples and descriptions

dcabib opened the pull request:
**Issue number:** #7119

## Summary

Enhances the Transfer Family parser models with field descriptions and examples using Pydantic's Field() functionality. This improvement provides better documentation and metadata for Transfer Family event parsing, following the pattern established in PR #7100.

### Changes

- Added Field() descriptions and examples to the Transfer Family parser model:
  - `TransferFamilyAuthorizer` - Lambda authorizer event model with comprehensive field documentation

- All field descriptions are based on official AWS Transfer Family Lambda authorizer documentation
- Examples sourced from actual test events (transferFamilyAuthorizer.json) and AWS documentation
- Maintains 100% backward compatibility - no breaking changes to type annotations or validation logic
- Preserves existing field aliases (serverId, sourceIp) and IPvAnyAddress type validation

### User experience

Customers will be able to see examples and descriptions when using Transfer Family parser models, providing:
- Rich IntelliSense with field descriptions and realistic examples
- Self-documenting code without needing external AWS documentation  
- Professional API documentation via .model_json_schema()
- Faster development with immediate reference for acceptable values
- Clear understanding of Transfer Family Lambda authorizer event structure

## Checklist

- [x] Meet tenets criteria
- [x] I have performed a self-review of this change
- [x] Changes have been tested (all existing tests pass)
- [x] Changes are documented (via Field descriptions)
- [x] PR title follows conventional commit semantics

**Is this a breaking change?** No

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

Closes #7119

dcabib committed:
refactor(parser): Improve Transfer Family models with examples and descriptions

Enhances the Transfer Family parser models with field descriptions and examples using Pydantic's Field() functionality. This improvement provides better documentation and metadata for Transfer Family event parsing, following the pattern established in PR #7100.

All field descriptions are based on official AWS Transfer Family Lambda authorizer documentation and include realistic examples from actual test events.

Closes #7119

dcabib committed:
security: fix password examples to avoid security hotspots

Replace specific password examples with placeholder format to resolve SonarQube security hotspots while maintaining documentation value.

leandrodamascena commented on aws_lambda_powertools/utilities/parser/models/transfer_family.py:
```diff
@@ -5,8 +5,28 @@
+        ...,
+        alias="sourceIp",
+        description="The IP address of the client connecting to the Transfer Family server.",
+        examples=["192.168.0.100", "10.0.0.50", "127.0.0.1", "203.0.113.12"],
```
Please remove examples in this case. It potentially will trigger some security pipeline in customers.

leandrodamascena reviewed (changes requested):
Hey @dcabib thanks for working on this. I left a comment.

dcabib committed:
security: remove IP address examples as requested

Remove IP address examples from source_ip field to avoid triggering security pipelines in customer environments as requested by leandrodamascena.

leandrodamascena reviewed (approved):
Thanks @dcabib! Approved.
